lib/amqp.js

```diff
@@ -112,7 +112,12 @@ class Amqp {
             log.error('WARNING - buffer full when publishing a message to '
                 + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
         }
-        return result ? result : eventToPromise(this.publishChannel, 'drain').then(() => true);
+
+        if (this.settings.PROCESS_AMQP_DRAIN) {
```

Comment:
@ElBabbler please do not use `drain` event but just switch to [ConfirmChannel](http://www.squaremobius.net/amqp.node/channel_api.html#confirmchannel)

Reply:
But, as I understud, ConfirmChannel can resolve 'acked' or 'nacked' message state only when it consumed, so It takes too much time, and requires sailor working?